src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -189,7 +189,9 @@ def _attn(self, query, key, value, attention_mask=None, head_mask=None):
         attn_weights = torch.matmul(query, key.transpose(-1, -2))
 
         if self.scale_attn_weights:
-            attn_weights = attn_weights / (value.size(-1) ** 0.5)
+            attn_weights = attn_weights / torch.tensor(value.size(-1) ** 0.5, dtype=attn_weights.dtype).to(
```

Comment:
So this will make `attn_weights` becomes FP32 in ONNX at the `where` OP?

Reply:
I feel like the situation is more like:

**without the fix:**
`attn_weights` will be in dtype fp32 (at where op) and `mask_value` as a constant always fp16

**with the fix:**
both in fp16

But please correct me as I know very little here (just by intuition)